### ISSUE TITLE
Codellama full finetune loss spkie

### ISSUE THREAD

**Issue by PoodleWang** [Lightning-AI/litgpt#628]
I used default configuration and my cuda is version is 11.8, torch is 2.0.dev20231002+cu118

fine-tune data: https://huggingface.co/datasets/theblackcat102/evol-codealpaca-v1
model: huggingface codellama-7b
No other code change except set devices = 8.  Running is simple, just python3 finetune/full.py
device is 8 A100 and one node. 

Here is the output: 

starting from certain iteration, the loss spikes.
iter 1151 step 144: loss 1.7019, iter time: 1106.53ms (optimizer.step)
iter 1152 step 144: loss 11.4630, iter time: 1260.82ms
iter 1153 step 144: loss 11.7545, iter time: 1109.35ms
iter 1154 step 144: loss 11.5019, iter time: 901.24ms
iter 1155 step 144: loss 11.4807, iter time: 1019.36ms
iter 1156 step 144: loss 12.1935, iter time: 978.51ms
iter 1157 step 144: loss 10.0220, iter time: 1022.02ms
iter 1158 step 144: loss 10.9364, iter time: 2538.55ms
iter 1159 step 145: loss 10.5513, iter time: 1628.55ms (optimizer.step)

I tried to other precision. It can go further but I the loss still get spike.
iter 3704 step 463: loss 4.7002, iter time: 897.30ms
iter 3705 step 463: loss 5.5483, iter time: 630.41ms
iter 3706 step 463: loss 2.3724, iter time: 685.74ms
iter 3707 step 463: loss 3.4107, iter time: 1399.17ms
iter 3708 step 463: loss 2.6863, iter time: 2032.91ms
iter 3709 step 463: loss 5.5921, iter time: 763.44ms
iter 3710 step 463: loss 2.8367, iter time: 678.69ms
iter 3711 step 464: loss 1.5079, iter time: 808.98ms (optimizer.step)
iter 3712 step 464: loss 9.3005, iter time: 673.26ms
iter 3713 step 464: loss 9.6611, iter time: 1371.52ms
iter 3714 step 464: loss 9.2431, iter time: 630.03ms
iter 3715 step 464: loss 9.7685, iter time: 709.66ms
iter 3716 step 464: loss 9.0485, iter time: 752.79ms
iter 3717 step 464: loss 8.9571, iter time: 714.44ms
iter 3718 step 464: loss 11.6645, iter time: 726.13ms
iter 3719 step 465: loss 10.1714, iter time: 852.62ms (optimizer.step)

**Comment by PoodleWang** [comment]
@carmocca  If you have time, please take a look at this error. This is really tricky, I even don't know how to debug it.

**Comment by PoodleWang** [comment]
Hi,  I lower the learning rate and also add the clip_norm. Now the trainning is stable (not quite sure, because the trainning is not finished).

**Comment by rasbt** [comment]
Sounds like this is resolved now! If not, please feel free to reopen and discuss! Also, we do have a [config_hub](https://github.com/Lightning-AI/litgpt/tree/main/config_hub/finetune) now. If you found a configuration that worked well and would like to contribute it, that'd be super welcome!
